xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -460,11 +465,30 @@ func securityConfigFromCommonTLSContextUsingNewFields(common *v3tlspb.CommonTlsC
 	}
 	// If we get here, it means that the `CertificateValidationContext` message
 	// was found through one of the supported ways. It is an error if the
-	// validation context is specified, but it does not contain the
-	// ca_certificate_provider_instance field which contains information about
-	// the certificate provider to be used for the root certificates.
+	// validation context is specified, but it does not specify a way to
+	// validate TLS certificates. Peer TLS certs can be verified in the
+	// following ways:
+	// 1. If the ca_certificate_provider_instance field is set, it contains
+	//    information about the certificate provider to be used for the root
+	//    certificates, else
+	// 2. If the system_root_certs field is set, and the config is for a client,
+	//    use the system default root certs.
 	if validationCtx.GetCaCertificateProviderInstance() == nil {
-		return nil, fmt.Errorf("expected field ca_certificate_provider_instance is missing in CommonTlsContext message: %+v", common)
+		systemRootCertsEnabled := envconfig.XDSSystemRootCertsEnabled && validationCtx.GetSystemRootCerts() != nil
+		// The `system_root_certs` field will not be supported on the gRPC
+		// server side. If `ca_certificate_provider_instance` is unset and
+		// `system_root_certs` is set, the LDS resource will be NACKed.
+		// - A82
+		if server && systemRootCertsEnabled {
+			return nil, fmt.Errorf("expected field ca_certificate_provider_instance is missing and unexpected field ca_certificate_provider_instance is set for server in CommonTlsContext message: %+v", common)
```

Comment:
This error message is quoting the same field `ca_certificate_provider_instance` twice. Needs fixing.